pareto/utilities/process_data.py

```diff
@@ -0,0 +1,932 @@
+                "Trucking arcs are given, but some trucking parameters are missing. The following missing parameters have been set to default values:"
+                + str(missing_trucking_data)
+            )
+            warnings.warn(warning_message, stacklevel=3)
+
+    # Set Default Data
+    # Iterate through all expected input. For all input left without data, fill with empty dictionary or default data.
+    # raise warning if empty dictionary or default is used.
+    default_used = []
+    # Defaults - Sets
+    all_set_input_tabs = get_valid_input_set_tab_names()
+
+    for set_tab in all_set_input_tabs:
+        if set_tab not in df_sets.keys():
+            df_sets[set_tab] = {}
+            default_used.append(set_tab)
+
+    # Defaults - Parameter
+    # Most default values are defined at the parameter definition. Some input data is not directly
+    # associated with a parameter (e.g. "Economics")
+    default_values = {
+        "Economics": {"discount_rate": 0.08, "CAPEX_lifetime": 20.0},
+    }
+    for input_tab in default_values.keys():
+        if input_tab not in df_parameters.keys():
+            df_parameters[input_tab] = default_values[input_tab]
+            default_used.append(input_tab)
+    # If an empty dictionary is passed to create_model(), the default value for the parameter
+    # is defined at the initialization of the parameter
+    all_parameter_input_tabs = get_valid_input_parameter_tab_names()
+    for param_tab in all_parameter_input_tabs:
+        if param_tab not in df_parameters.keys():
+            df_parameters[param_tab] = {}
+            default_used.append(param_tab)
+    # Raise warning for default values
+    if len(default_used) > 0:
+        warning_message = (
+            f"The following parameters were missing and default values were substituted:"
+            + str(default_used)
+        )
+        warnings.warn(warning_message, stacklevel=3)
+    return (df_sets, df_parameters)
+
+
+def model_infeasibility_detection(strategic_model):
+    # check_required_data() performs basic feasibility checks:
+    # - at least one source node in the model
+    # - at least on sink node in the model
+    # - each source node has an outgoing arc, each sink node has an incoming arc, and all other nodes have both an incoming and an outgoing arc.
+
+    capacity_feasibility_message = []
+    demand_feasibility_message = []
+    # Checks that for each time period, the volume of produced water is below the maximum capacity limit of sink nodes
+    # Get the total system produced water for each time period (PadRates, FLowbackRates)
+    strategic_model.p_beta_TotalPW = Param(
+        strategic_model.s_T,
+        units=strategic_model.model_units["volume"],
+        doc="Combined water supply forecast (flowback & production) over the planning horizon [volume]",
+        mutable=True,  # Mutable Param - can be changed in sensitivity analysis without rebuilding the entire model
+    )
```

Comment:
Why is this implemented as a mutable parameter? I don't see any benefit to using a mutable `Param` over just using a Python variable. You could use a Pyomo expression to do the calculation - I think that would be sensible.

This comment applies to all of the `Params` which are defined in ``model_infeasibility_detection().

Reply:
Reasonable point - majority of our Params in PARETO are mutable, so I started here, but I see your point. These parameters are dependent on other params (e.g. defined by expressions), so they should not separately be mutable. Updated.

Reply:
This looks good to me with the latest commits. Resolving.